util/network-devp2p/src/node_table.rs

```diff
@@ -122,18 +123,31 @@ impl FromStr for NodeEndpoint {
 				address: a,
 				udp_port: a.port()
 			}),
-			Ok(_) => Err(ErrorKind::AddressResolve(None).into()),
-			Err(e) => Err(ErrorKind::AddressResolve(Some(e)).into())
+			Ok(None) => {
+				// REVIEW: how is this case possible? I think we can remove this case.
+				Err(ErrorKind::AddressResolve(None).into())
+			},
+			Err(e) => {
+				match e.kind() {
+					io::ErrorKind::InvalidInput => {
+						Err(ErrorKind::AddressParse(Some(e)).into())
+					},
+					_ => {
+						Err(e.into())
```

Comment:
this will be transformed into `Error::Io(...)`. Is it desired?

Reply:
How about we return `Err(ErrorKind::AddressParse())` for all error cases here? In light of your other comment below and https://doc.rust-lang.org/src/std/net/addr.rs.html#891 it seems kind of silly to cover other cases (there aren't any, `to_socket_addrs` always only return `InvalidInput`). This match arm would become:

```rust
Err(e) => Err(ErrorKind::AddressParse())
```

Thoughts?